src/Tasks/ResourceHandling/MSBuildResXReader.cs

```diff
@@ -0,0 +1,300 @@
+
+        private static void ParseAssemblyAlias(Dictionary<string,string> aliases, XElement elem)
+        {
+            string alias = elem.Attribute("alias")?.Value;
+            string name = elem.Attribute("name").Value;
+
+            if (string.IsNullOrEmpty(alias))
+            {
+                AssemblyName assemblyName = new AssemblyName(name);
+
+                alias = assemblyName.Name;
+            }
+
+            // Match original last-alias-definition-wins behavior
+            // https://github.com/dotnet/winforms/blob/33b9fe202f3dc1b8e7c4bf28492f8bd3252f1a20/src/System.Windows.Forms/src/System/Resources/ResXResourceReader.cs#L732-L738
+            aliases[alias] = name;
+        }
+
+        // Consts from https://github.com/dotnet/winforms/blob/16b192389b377c647ab3d280130781ab1a9d3385/src/System.Windows.Forms/src/System/Resources/ResXResourceWriter.cs#L46-L63
+        private const string Beta2CompatSerializedObjectMimeType = "text/microsoft-urt/psuedoml-serialized/base64";
+        private const string CompatBinSerializedObjectMimeType = "text/microsoft-urt/binary-serialized/base64";
+        private const string CompatSoapSerializedObjectMimeType = "text/microsoft-urt/soap-serialized/base64";
+        private const string BinSerializedObjectMimeType = "application/x-microsoft.net.object.binary.base64";
+        private const string SoapSerializedObjectMimeType = "application/x-microsoft.net.object.soap.base64";
+        private const string DefaultSerializedObjectMimeType = BinSerializedObjectMimeType;
+        private const string ByteArraySerializedObjectMimeType = "application/x-microsoft.net.object.bytearray.base64";
+        private const string ResMimeType = "text/microsoft-resx";
+
+        private const string StringTypeName = "System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089";
+
+        private static string GetFullTypeNameFromAlias(string aliasedTypeName, Dictionary<string, string> aliases)
+        {
+            if (aliasedTypeName == null)
+            {
+                return StringTypeName;
+            }
+
+            int indexStart = aliasedTypeName.IndexOf(',');
+            if (aliases.TryGetValue(aliasedTypeName.Substring(indexStart + 2), out string fullAssemblyIdentity))
+            {
+                return aliasedTypeName.Substring(0, indexStart + 2) + fullAssemblyIdentity;
+            }
+
+            // Allow "System.String" bare
+            if (aliasedTypeName.Equals("System.String", StringComparison.Ordinal))
+            {
+                return StringTypeName;
+            }
+
+            // No alias found. Hope it's sufficiently complete to be resolved at runtime
+            return aliasedTypeName;
+        }
+
+        private static void ParseData(string resxFilename, bool pathsRelativeToBasePath, List<IResource> resources, Dictionary<string,string> aliases, XElement elem)
+        {
+            string name = elem.Attribute("name").Value;
```

Comment:
Who checks that the xml is in the right resx format?

Reply:
Yeah, I struggled with going strict or permissive here. The current error handling approach is "assume good faith and throw". I'm optimistic that that's ok since most of these files are machine-generated and machine-edited. If it proves not to be in the previews we can make the errors much nicer.

Reply:
@BenVillalobos and I came up with a cheap-but-better-than-it-was improvement in 442c03b. I think that's good enough for 16.3p1.

Reply:
I'm not a fan of that catch (Exception). If I followed right, it will lead to confounding user-facing errors like "MSB3103: Invalid Resx file. Object reference not set to an instance of an object.`" when there is actually a bug in our code.

Reply:
On this being good enough for 16.3p1, my experience is that we don't go back and make aggressive catches less aggressive. I'd rather leave proper error handling unfinished than hacked for a preview.